src/gsumm.c

```diff
@@ -1037,9 +1061,19 @@ SEXP glast(SEXP x) {
       if (isunsorted) k = oo[k]-1;
       k = (irowslen == -1) ? k : irows[k]-1;
       dans[i] = dx[k];
+      if (narm){
```

Comment:
A minor thing is to align to the coding style convention of a file you are editing. As you can see around, add extra space between in cases like:
- `if (narm){` -> `if (narm) {`
- `while(...){` -> `while (...) {`